### PR TITLE
test(mantle): Op-level proof pairing and gas tests

### PR DESCRIPTION
**Pull request by strinnityk**
## 1. What does this PR implement?

Add op-to-proof pairings tests.

Add execution gas tests for every op, covering threshold scaling and overflow.

Add `From` conversions from each operation into `Op`, and getters for the parts of `SignedOpError`.

Drops a commented-out `BinaryDecode for OpProof`.

## 2. Does the code have enough context to be clearly understood?
Yes

## 3. Who are the specification authors and who is accountable for this PR?
@strinnityk

## 4. Is the specification accurate and complete?
N/A

## 5. Does the implementation introduce changes in the specification?
N/A

## Checklist

> [!WARNING]
> Do not merge the PR if any of the following is missing:

* [X] 1. The PR title follows the Conventional Commits [specification](https://www.notion.so/How-to-open-a-Pull-Request-215261aa09df80deb538ed59f5e4e0b5).
* [X] 2. Description added.
* [X] 3. Context and links to Specification document(s) added.
* [X] 4. Main contact(s) (developers and specification authors) added
* [X] 5. Implementation and Specification are 100% in sync including changes. This is critical.
* [X] 6. Link PR to a specific milestone.
* [X] 7. New or changed logs were reviewed against the [logging guidelines](https://github.com/logos-blockchain/logos-blockchain/blob/master/CONTRIBUTING.md#logging-guidelines).

---

<sub>Stack created with <a href="https://github.com/github/gh-stack">GitHub Stacks CLI</a> • <a href="https://gh.io/stacks-feedback">Give Feedback 💬</a></sub>

### CODE REVIEW

**Review by youngjoon-lee — APPROVED** [review]

File: core/src/mantle/ops/op.rs [hunk 1]
```diff
@@ -195,3 +195,61 @@ impl Op {
...
+                Op::ChannelDeposit(_) => |proof| matches!(proof, OpProof::ZkSig(_)),
+                Op::ChannelWithdraw(_) => |proof| matches!(proof, OpProof::ChannelMultiSigProof(_)),
+                Op::ChannelTransfer(_) => |proof| matches!(proof, OpProof::ChannelMultiSigProof(_)),
+                Op::SDPDeclare(_) => |proof| matches!(proof, OpProof::ZkAndEd25519Sigs(_)),
+                Op::SDPWithdraw(_) => |proof| matches!(proof, OpProof::ZkSig(_)),
+                Op::SDPActive(_) => |proof| matches!(proof, OpProof::ZkSig(_)),
+                Op::LeaderClaim(_) => |proof| matches!(proof, OpProof::PoC(_)),
+                Op::Transfer(_) => |proof| matches!(proof, OpProof::ZkSig(_)),
+                Op::ClaimPowReward(_) => |proof| matches!(proof, OpProof::None(_)),
+            };
```
Comment (danielSanchezQ) [inline core/src/mantle/ops/op.rs:244]: Maybe this should be its own method, at least defined outside of the loop?

**Review by danielSanchezQ — APPROVED** [review]
